ui/app/AppLayouts/Wallet/SendModal.qml

```diff
@@ -1,27 +1,188 @@
 import QtQuick 2.13
 import QtQuick.Controls 2.13
 import QtQuick.Layouts 1.13
+import QtQuick.Dialogs 1.3
 import "../../../imports"
 import "../../../shared"
 import "./components"
 
 ModalPopup {
-    id: popup
+    id: root
 
     //% "Send"
     title: qsTrId("command-button-send")
     height: 700
 
-    onOpened: {
-        sendModalContent.amountInput.selectedAmount = ""
-        sendModalContent.passwordInput.text = ""
-        sendModalContent.amountInput.forceActiveFocus(Qt.MouseFocusReason)
+    property MessageDialog sendingError: MessageDialog {
+        id: sendingError
+        title: qsTr("Error sending the transaction")
+        icon: StandardIcon.Critical
+        standardButtons: StandardButton.Ok
     }
+    property MessageDialog sendingSuccess: MessageDialog {
+        id: sendingSuccess
+        //% "Success sending the transaction"
+        title: qsTrId("success-sending-the-transaction")
+        icon: StandardIcon.NoIcon
+        standardButtons: StandardButton.Ok
+        onAccepted: {
+            root.close()
+        }
+    }
+
+    onClosed: {
+        stack.reset()
+    }
+
+    function sendTransaction() {
+        let responseStr = walletModel.sendTransaction(selectFromAccount.selectedAccount.address,
+                                                 selectRecipient.selectedRecipient.address,
+                                                 txtAmount.selectedAsset.address,
+                                                 txtAmount.selectedAmount,
+                                                 gasSelector.selectedGasLimit,
+                                                 gasSelector.selectedGasPrice,
+                                                 transactionSigner.enteredPassword)
+        let response = JSON.parse(responseStr)
 
-    SendModalContent {
-        id: sendModalContent
-        closePopup: function () {
-            popup.close()
+        if (response.error) {
+            // It's an error
+            if (response.error.includes("could not decrypt key with given password")){
+                transactionSigner.validationError = qsTr("Wrong password")
+                return
+            }
+            sendingError.text = response.error
+            return sendingError.open()
+        }
+
+        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2/%1").arg(response.result).arg(walletModel.etherscanLink)
+        sendingSuccess.open()
+    }
+
+    TransactionStackView {
+        id: stack
+        height: parent.height
```

Comment:
This line is not necessary with `anchors.fill: parent`

Reply:
Removed.